File: aoc/aoc21/python/day1.py
def readFile(filepath):
    with open(filepath, "r") as f:
        return [int(line) for line in f.readlines()]

def solve(depths):
    ups = [x for x in zip(depths, depths[1:]) if x[1] > x[0]]
    return len(ups)

def solve2(depths):
    groups = [x[0]+x[1]+x[2] for x in zip(depths, depths[1:], depths[2:])]
    return solve(groups)

depths = readFile('./inputs/day1_example.txt')
print("day 1 part 1 example = " + str(solve(depths)))
print("day 1 part 2 example = " + str(solve2(depths)))

depths = readFile('./inputs/day1.txt')
print("day 1 part 1 = " + str(solve(depths)))
print("day 1 part 2 = " + str(solve2(depths)))

File: aoc/aoc22/js/day1.js
const fs = require('fs');

function readFromFile(filename)
{
    const fileData = fs.readFileSync(filename, 'utf8');
    const lines = fileData.split(/\n/);

    const elves = []

    let elf = [];
    lines.forEach(line => {
        line = line.trim();
        if (!line){
            if (elf.length > 0){
                elves.push(elf);
                elf = [];
            }
        } else {
            elf.push(Number.parseInt(line));
        }
    });
    if (elf.length > 0){
        elves.push(elf);
    }
    return elves;
}

function sumElves(elves)
{
    const sums = [];
    elves.forEach(elf => {
        sums.push(elf.reduce((a, b) => a + b, 0));
    });
    return sums;
}

let elves = readFromFile('../inputs/day1_example.txt');
let sum = sumElves(elves);
sum.sort((a, b) => b - a);

console.log("day 1 part 1 example = " + sum[0]);
console.log("day 1 part 2 example = " + (sum[0] + sum[1] + sum[2]));

elves = readFromFile('../inputs/day1.txt');
sum = sumElves(elves);
sum.sort((a, b) => b - a);
console.log("day 1 part 1 = " + sum[0]);
console.log("day 1 part 2 = " + (sum[0] + sum[1] + sum[2]));
